### ISSUE TITLE
[java] ExcessiveImports shouldn't count classes from testing libraries

### ISSUE THREAD

**Issue by php-coder** [pmd/pmd#4036]
**Affects PMD Version:** 
6.44.0

**Rule:**
https://pmd.github.io/latest/pmd_rules_java_design.html#excessiveimports

**Description:**
At this moment `ExcessiveImports` takes into account imports of the classes from testing libraries (jUnit, Mockito, AssertJ). Tests might contains a lot of such imports and that triggers a violation. The rule is supposed to help developers to identify coupling **within their own classes** and testing tools doesn't affect on that as they are used as helpers within test classes only.

I suggest to exclude such classes from counting.

**Code Sample demonstrating the issue:**
```java
import org.assertj.core.api.SoftAssertions;
import org.junit.jupiter.api.AfterEach;
import org.junit.jupiter.api.Assertions;
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Test;
import org.junit.jupiter.api.extension.ExtensionContext;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.ArgumentsProvider;
import org.junit.jupiter.params.provider.ArgumentsSource;
import org.junit.jupiter.params.provider.CsvSource;
import org.junit.jupiter.params.provider.ValueSource;
import org.mockito.Mock;
import org.mockito.Mockito;
import org.mockito.MockitoAnnotations;
[12 imports of own classes]

import java.util.List;
import java.util.Optional;
import java.util.stream.Stream;

import static org.junit.jupiter.api.Assertions.assertEquals;
import static org.junit.jupiter.params.provider.Arguments.arguments;
```

**Expected outcome:**
PMD reports a violation at a whole file, but that's wrong. That's a false positive.

**Running PMD through:** *Gradle*

**Comment by is-ivanov** [comment]
This improve is very usefull

> 
> I suggest to provide a property for the rule to identify packages from which to exclude class imports from counting.
>
